File: app/src/main/java/com/example/hitchapp/repositories/UserRepository.java
package com.example.hitchapp.repositories;

import com.example.hitchapp.models.User;
import com.parse.LogInCallback;
import com.parse.ParseFile;
import com.parse.ParseUser;
import com.parse.SaveCallback;
import com.parse.SignUpCallback;

public class UserRepository {

    public static final String TAG = "UserRepository";
    private static UserRepository instance;

    public static UserRepository getInstance(){
        if(instance == null){
            instance = new UserRepository();
        }
        return instance;
    }

    public void loginUser(String username, String password, LogInCallback loginCallBack){
        ParseUser.logInInBackground(username, password, loginCallBack);
    }

    public void signupUser(User user, SignUpCallback signUpCallback){
        user.signUpInBackground(signUpCallback);
    }

    public void savePhoto(ParseFile photo, SaveCallback saveCallback){
        photo.saveInBackground(saveCallback);
    }
}

File: app/src/main/java/com/example/hitchapp/repositories/UserRepository.kt
package com.example.hitchapp.repositories

import com.example.hitchapp.models.User
import com.parse.*

class UserRepository {
    fun loginUser(username: String?, password: String?, loginCallBack: LogInCallback?) {
        ParseUser.logInInBackground(username, password, loginCallBack)
    }

    fun signupUser(user: User, signUpCallback: SignUpCallback?) {
        user.signUpInBackground(signUpCallback)
    }

    fun savePhoto(photo: ParseFile, saveCallback: SaveCallback?) {
        photo.saveInBackground(saveCallback)
    }

    companion object {
        const val TAG = "UserRepository"
        var instance: UserRepository? = null
            get() {
                if (field == null) {
                    field = UserRepository()
                }
                return field
            }
            private set
    }
}